tests/model_serving/model_runtime/vllm/conftest.py

```diff
@@ -0,0 +1,133 @@
+):
+    accelerator_type = supported_accelerator_type.lower()
+    template_name = TEMPLATE_MAP.get(accelerator_type, "vllm-runtime-template")
+    manifest = get_runtime_manifest(
+        client=admin_client,
+        template_name=template_name,
+        deployment_type=request.param["deployment_type"],
+        runtime_image=vllm_runtime_image,
+    )
+    manifest["metadata"]["name"] = "vllm-runtime"
+    manifest["metadata"]["namespace"] = model_namespace.name
+    with ServingRuntime(client=admin_client, kind_dict=manifest) as model_runtime:
+        yield model_runtime
+
+
+@pytest.fixture(scope="session")
+def skip_if_no_supported_accelerator_type(supported_accelerator_type: str):
+    if not supported_accelerator_type:
+        pytest.skip("Accelartor type is not provide,vLLM test can not be run on CPU")
+
+
+@pytest.fixture(scope="class")
+def vllm_inference_service(
+    request: FixtureRequest,
+    admin_client: DynamicClient,
+    model_namespace: Namespace,
+    serving_runtime: ServingRuntime,
+    supported_accelerator_type: str,
+    ci_s3_storage_uri: str,
+    model_service_account: ServiceAccount,
+) -> Generator[InferenceService, Any, Any]:
+    isvc_kwargs = {
+        "client": admin_client,
+        "name": request.param["name"],
+        "namespace": model_namespace.name,
+        "runtime": serving_runtime.name,
+        "storage_uri": ci_s3_storage_uri,
+        "model_format": serving_runtime.instance.spec.supportedModelFormats[0].name,
+        "model_service_account": model_service_account.name,
+        "deployment_mode": request.param.get("deployment-mode", KServeDeploymentType.SERVERLESS),
+    }
+    accelerator_type = supported_accelerator_type.lower()
+    gpu_count = request.param.get("gpu_count")
+    identifier = ACCELERATOR_IDENTIFIER.get(accelerator_type, "nvidia.com/gpu")
+    resources = PREDICT_RESOURCES["resources"]
+    resources["requests"][identifier] = gpu_count
+    resources["limits"][identifier] = gpu_count
+    isvc_kwargs["resources"] = resources
+
+    if gpu_count > 1:
+        isvc_kwargs["volumes"] = PREDICT_RESOURCES["volumes"]
+        isvc_kwargs["volumes_mounts"] = PREDICT_RESOURCES["volume_mounts"]
+    if arguemnts := request.param.get("runtime_argument"):
+        arguemnts.append(f"--tensor-parallel-size={gpu_count}")
+        isvc_kwargs["argument"] = arguemnts
+
+    if min_replicas := request.param.get("min-replicas"):
+        isvc_kwargs["min_replicas"] = min_replicas
+
+    with create_isvc(**isvc_kwargs) as isvc:
+        yield isvc
+
+
+@pytest.fixture(scope="class")
+def model_service_account(admin_client: DynamicClient, kserve_endpoint_s3_secret: Secret):
+    with ServiceAccount(
+        client=admin_client,
+        namespace=kserve_endpoint_s3_secret.namespace,
+        name="models-bucket-sa",
+        secrets=[{"name": kserve_endpoint_s3_secret.name}],
+    ) as sa:
+        yield sa
+
+
+@pytest.fixture(scope="class")
+def kserve_endpoint_s3_secret(
```

Comment:
Can't you reuse the `endpoint_s3_secret` fixture in tests/model_serving/model_server/private_endpoint/conftest.py?

Reply:
It seems like you're still using the same fixture, did you mix up with the earlier comment?

Reply:
yes, it seems that..to be frank i don't want to use that because of the reason i mentioned in the other comment. That is why i created this fixture in runtime